packages/sitecore-jss-nextjs/src/revalidate/revalidate-middleware.ts

```
@@ -0,0 +1,256 @@
+    const filteredUpdates = this.getFilteredUpdates(req);
+
+    if (this.isEmpty(filteredUpdates)) {
+      // nothing to revalidate
+      return res.status(204).json({});
+    }
+
+    // extract only paths from filtered updates object
+    const paths = this.extractPaths(filteredUpdates);
+
+    const pathsToRevalidate: string[] = [];
+
+    // when multiSite and personalization both are configured or only personalization is configured
+    if (this.config.personalize) {
+      const personalizeInfo = await this.getPersonalizedResults(filteredUpdates);
+      const personalizedRewritePaths = this.getRewritePaths(
+        personalizeInfo.personalized,
+        this.config.multiSite
+      );
+      const nonPersonalizedRewritePaths = this.getRewritePaths(
+        personalizeInfo.nonPersonalized,
+        this.config.multiSite
+      );
+      pathsToRevalidate.push(...personalizedRewritePaths, ...nonPersonalizedRewritePaths);
+    }
+
+    // when only multiSite is configured but personalization is not
+    if (this.config.multiSite && !this.config.personalize) {
+      const multiSitePaths = paths.map((path: string) => `/${SITE_PREFIX}${path}`);
+      pathsToRevalidate.push(...multiSitePaths);
+    }
+
+    // when both multisite and personalization are not configured
+    if (!this.config.multiSite && !this.config.personalize) {
+      const defaultPaths = paths.map((path: string) => this.getPathName(path));
+      pathsToRevalidate.push(...defaultPaths);
+    }
+
+    try {
+      // revalidate all the collected paths
+      await Promise.all(pathsToRevalidate.map((path: string) => res.revalidate(path)));
+      console.log(`[api/revalidate] revalidate: ${pathsToRevalidate.join(', ')}`);
+      return res.status(200).json({ revalidated: true });
+    } catch (error) {
+      console.error(`[api/revalidate] error: ${error}`);
+      return res.status(500).json({ message: 'Error revalidating' });
```

Do we want to return an error message in a response?

---

Are you suggesting to return the error response or return an empty json?